services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/DiagnosisKeyBundler.java

```diff
@@ -163,22 +174,44 @@ protected LocalDateTime getSubmissionDateTime(DiagnosisKey diagnosisKey) {
   }
 
   /**
-   * Returns all diagnosis keys that should be distributed on a specific date.
+   * Returns all diagnosis keys that should be distributed on a specific date for a specific country.
    */
-  public List<DiagnosisKey> getDiagnosisKeysForDate(LocalDate date) {
-    return this.distributableDiagnosisKeys.keySet().stream()
+  public List<DiagnosisKey> getDiagnosisKeysForDate(LocalDate date, String country) {
+    if (!supportedCountries.contains(country)) {
```

Comment:
This validation could be extracted to its own method as there are multiple occurences

Reply:
Also, the runtime exception causes the distribution process to be killed. Is this the intended approach for dealing with such scenarios where keys from unkown countries are "requested" from the bundler? Not sure if that can happen or not though.

Reply:
I would rather log a warning and return an empty list